lib/charms/hydra/v0/oauth.py

```diff
@@ -201,11 +202,29 @@ def _dump_data(data: Dict, schema: Optional[Dict] = None) -> Dict:
                 ret[k] = json.dumps(v)
             except json.JSONDecodeError as e:
                 raise DataValidationError(f"Failed to encode relation json: {e}")
+        elif isinstance(v, bool):
+            ret[k] = str(v)
         else:
             ret[k] = v
     return ret
 
 
+def strtobool(val: str) -> bool:
+    """Convert a string representation of truth to true (1) or false (0).
+
+    True values are 'y', 'yes', 't', 'true', 'on', and '1'; false values
+    are 'n', 'no', 'f', 'false', 'off', and '0'.  Raises ValueError if
+    'val' is anything else.
+    """
+    val = val.lower()
```

Comment:
suggestion(non-blocking): careful here since you're assuming that `val` is never `None` (or this `.lower()` would raise an error), even though I've seen you perform checks outside of this func. Could be better to handle None inside the func mainly to avoid misuse/surprises

Reply:
Fair